### ISSUE TITLE
interface slow when file flow is processing

### ISSUE THREAD

**Issue by spupuz** [revenz/FileFlows#392]
i noticed a really slow interface when fileflow is processing files.
i use docker with vaapi quick sync on a debian server have around 7000 files mixed audio and video.

got this errors on log:

```
2023-01-13 07:53:14.784 [ERRR] -> Took '00:01:19.7172419' to update object
2023-01-13 07:53:14.919 [ERRR] -> Took '00:01:19.1793898' to update object
2023-01-13 07:53:15.096 [ERRR] -> Took '00:01:19.2631996' to update object
2023-01-13 07:53:15.268 [ERRR] -> Took '00:01:19.3286937' to update object
2023-01-13 07:53:16.172 [ERRR] -> Took '00:01:20.1363345' to update object
2023-01-13 07:53:16.275 [ERRR] -> Took '00:01:20.1358047' to update object
2023-01-13 07:53:16.360 [ERRR] -> Took '00:01:19.9090093' to update object
2023-01-13 07:53:16.538 [ERRR] -> Took '00:01:17.7508555' to update object
in log
and also
7:53:50.786 [ERRR] -> Took '00:01:38.1135813' to execute: select Status from LibraryFile where Uid = '2db842dd-b5aa-4cde-8f1b-a62498b1a183'
```

**Comment by revenz** [comment]
can you send me the full logs
- server
- db log
- file log, but if this is happening when nothing is processing i dont need this
- processing node, see above, and only if using a separate processing node

and your sqlite db file.

first I need to reproduce this

**Comment by spupuz** [comment]
server:  Docker Version: [1.0.8.2176]
db log:  where is db log?
file log, but if this is happening when nothing is processing i dont need this: he are logs
[FileFlows-Jan14-01 (3).log](https://github.com/revenz/FileFlows/files/10418913/FileFlows-Jan14-01.3.log)
[FileFlows-Jan14-01 (2).log](https://github.com/revenz/FileFlows/files/10418915/FileFlows-Jan14-01.2.log)
[FileFlows-Jan14-01 (1).log](https://github.com/revenz/FileFlows/files/10418916/FileFlows-Jan14-01.1.log)
[FileFlows-Jan14-01.log](https://github.com/revenz/FileFlows/files/10418917/FileFlows-Jan14-01.log)
processing node, see above, and only if using a separate processing node: i only use internal node

does db log contain password in clear?

**Comment by revenz** [comment]
so is this only happening when files are processing???

I'm trying to test on a 7500 file system using sqlite and havent been able to reproduce this yet

**Comment by spupuz** [comment]
mostly when processing video files. not audio mp3



Il giorno sab 14 gen 2023 alle ore 23:03 John Andrews <
***@***.***> ha scritto:

**Comment by spupuz** [comment]
```
 neofetch
       _,met$$$$$gg.          root@openmediavault
    ,g$$$$$$$$$$$$$$$P.       -------------------
  ,g$$P"     """Y$$.".        OS: Debian GNU/Linux 11 (bullseye) x86_64
 ,$$P'              `$$$.     Kernel: 5.15.83-1-pve
',$$P       ,ggs.     `$$b:   Uptime: 6 days, 6 hours, 3 mins
`d$$'     ,$P"'   .    $$$    Packages: 788 (dpkg)
 $$P      d$'     ,    $$P    Shell: bash 5.1.4
 $$:      $$.   -    ,d$$'    Resolution: 1920x1080
 $$;      Y$b._   _,d$P'      Terminal: /dev/pts/0
 Y$$.    `.`"Y$$$$P"'         CPU: Intel i7-5775C (8) @ 3.700GHz
 `$$b      "-.__              GPU: Intel Iris Pro Graphics 6200
  `Y$$                        Memory: 24386MiB / 31792MiB
   `Y$$.
     `$$b.
       `Y$$b.
          `"Y$b._
              `"""
```

**Comment by spupuz** [comment]
Problems seems to kick in when scanning + reprocessing

**Comment by revenz** [comment]
finally able to reproduce this

```
2023-01-29 20:47:43.566 -> REQUEST [POST] [200]: /api/worker/work/update
2023-01-29 20:47:43.598 -> REQUEST [POST] [200]: /api/worker/work/update
2023-01-29 20:47:43.629 -> REQUEST [POST] [200]: /api/worker/work/update
2023-01-29 20:47:43.660 -> REQUEST [POST] [200]: /api/worker/work/update
2023-01-29 20:47:43.695 -> REQUEST [POST] [200]: /api/worker/work/update
2023-01-29 20:47:43.729 -> REQUEST [POST] [200]: /api/worker/work/update
2023-01-29 20:47:43.761 -> REQUEST [POST] [200]: /api/worker/work/update
2023-01-29 20:47:43.804 -> REQUEST [POST] [200]: /api/worker/work/update
2023-01-29 20:47:43.849 -> REQUEST [POST] [200]: /api/worker/work/update
2023-01-29 20:47:43.881 -> REQUEST [POST] [200]: /api/worker/work/update
2023-01-29 20:47:43.912 -> REQUEST [POST] [200]: /api/worker/work/update
2023-01-29 20:47:43.942 -> REQUEST [POST] [200]: /api/worker/work/update
2023-01-29 20:47:43.976 -> REQUEST [POST] [200]: /api/worker/work/update
2023-01-29 20:47:44.007 -> REQUEST [POST] [200]: /api/worker/work/update
2023-01-29 20:47:44.038 -> REQUEST [POST] [200]: /api/worker/work/update
2023-01-29 20:47:44.069 -> REQUEST [POST] [200]: /api/worker/work/update
2023-01-29 20:47:44.100 -> REQUEST [POST] [200]: /api/worker/work/update
2023-01-29 20:47:44.131 -> REQUEST [POST] [200]: /api/worker/work/update
2023-01-29 20:47:44.161 -> REQUEST [POST] [200]: /api/worker/work/update
2023-01-29 20:47:44.192 -> REQUEST [POST] [200]: /api/worker/work/update
2023-01-29 20:47:44.223 -> REQUEST [POST] [200]: /api/worker/work/update
2023-01-29 20:47:44.259 -> REQUEST [POST] [200]: /api/worker/work/update
2023-01-29 20:47:44.291 -> REQUEST [POST] [200]: /api/worker/work/update
2023-01-29 20:47:44.324 -> REQUEST [POST] [200]: /api/worker/work/update
2023-01-29 20:47:44.357 -> REQUEST [POST] [200]: /api/worker/work/update
2023-01-29 20:47:44.395 -> REQUEST [POST] [200]: /api/worker/work/update
2023-01-29 20:47:44.426 -> REQUEST [POST] [200]: /api/worker/work/update
2023-01-29 20:47:44.459 -> REQUEST [POST] [200]: /api/worker/work/update
2023-01-29 20:47:44.490 -> REQUEST [POST] [200]: /api/worker/work/update
2023-01-29 20:47:44.522 -> REQUEST [POST] [200]: /api/worker/work/update
2023-01-29 20:47:44.554 -> REQUEST [POST] [200]: /api/worker/work/update
2023-01-29 20:47:44.585 -> REQUEST [POST] [200]: /api/worker/work/update
2023-01-29 20:47:44.615 -> REQUEST [POST] [200]: /api/worker/work/update
2023-01-29 20:47:44.646 -> REQUEST [POST] [200]: /api/worker/work/update
2023-01-29 20:47:44.734 -> REQUEST [POST] [200]: /api/worker/work/update
2023-01-29 20:47:44.767 -> REQUEST [POST] [200]: /api/worker/work/update
2023-01-29 20:47:44.812 -> REQUEST [POST] [200]: /api/worker/work/update
2023-01-29 20:47:44.844 -> REQUEST [POST] [200]: /api/worker/work/update
2023-01-29 20:47:44.876 -> REQUEST [POST] [200]: /api/worker/work/update
2023-01-29 20:47:44.908 -> REQUEST [POST] [200]: /api/worker/work/update
2023-01-29 20:47:44.939 -> REQUEST [POST] [200]: /api/worker/work/update
2023-01-29 20:47:44.975 -> REQUEST [POST] [200]: /api/worker/work/update
2023-01-29 20:47:45.006 -> REQUEST [POST] [200]: /api/worker/work/update
2023-01-29 20:47:45.038 -> REQUEST [POST] [200]: /api/worker/work/update
2023-01-29 20:47:45.069 -> REQUEST [POST] [200]: /api/worker/work/update
2023-01-29 20:47:45.100 -> REQUEST [POST] [200]: /api/worker/work/update
2023-01-29 20:47:45.134 -> REQUEST [POST] [200]: /api/worker/work/update
2023-01-29 20:47:45.166 -> REQUEST [POST] [200]: /api/worker/work/update
2023-01-29 20:47:45.198 -> REQUEST [POST] [200]: /api/worker/work/update
2023-01-29 20:47:45.231 -> REQUEST [POST] [200]: /api/worker/work/update
2023-01-29 20:47:45.268 -> REQUEST [POST] [200]: /api/worker/work/update
2023-01-29 20:47:45.301 -> REQUEST [POST] [200]: /api/worker/work/update
2023-01-29 20:47:45.332 -> REQUEST [POST] [200]: /api/worker/work/update
```
issue seems to be a lot of constant small updates requiring the db to be updated many times per second.

solution is to limit the updates

**Comment by spupuz** [comment]
Cool will you release a fix soon?

**Comment by revenz** [comment]
will be a little while, this doesnt warrant an urgent update.  will be included in 1.0.9, which has no ETA atm.

**Comment by revenz** [comment]
file updates are now limited to an update every 2 seconds, or at a step change.  this reduces load on the server and improves performance
